src/prover_disk.hpp

```diff
@@ -122,18 +120,15 @@ class DiskProver {
     ~DiskProver()
     {
         std::lock_guard<std::mutex> l(_mtx);
-        delete[] this->memo;
         for (int i = 0; i < 6; i++) {
             Encoding::ANSFree(kRValues[i]);
         }
         Encoding::ANSFree(kC3R);
     }
 
-    void GetMemo(uint8_t* buffer) { memcpy(buffer, memo, this->memo_size); }
-
-    uint32_t GetMemoSize() const noexcept { return this->memo_size; }
+    std::vector<uint8_t> GetMemo() { return memo; }
```

Comment:
would it make sense for this to return a `span<const uint8_t>` or a `const std::vector<uint8_t>&`? (as well as `GetId()` below)

Reply:
I agree `span` would be nice in many places but we have C++17 set here still so this feels out of scope. I changed the return to const reference though :+1: